server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -301,6 +301,18 @@ public class IndicesService extends AbstractLifecycleComponent
         Property.Final
     );
 
+    /**
+     * This setting is used to prevents creation of indices where the 'index.replication.type' setting does not match
+     * with the replication type index setting, set at cluster level i.e. 'cluster.indices.replication.strategy'.
+     * If disabled, the replication type can be specified.
+     */
+    public static final Setting<Boolean> CLUSTER_FORCE_INDEX_REPLICATION_TYPE_SETTING = Setting.boolSetting(
+        "cluster.force.index.replication_type",
```

Comment:
The PR description, issue, and documentation issue all mention `cluster.restrict.index.replication_type`. This is just a note to ensure those get updated with whatever is decided here.

`cluster.remote_store.index.restrict.async-durability` is the existing example I can find where "restrict" is used. Why use "force" here?

Using `cluster.indices.replication.strategy` as a precedent, this should probably replace `replication_type` with `replication.type`.

Reply:
>The PR description, issue, and documentation issue all mention cluster.restrict.index.replication_type. This is just a note to ensure those get updated with whatever is decided here.

Thanks for the callout. I updated PR description, issue and linked documentation issue to use `cluster.force.index.replication.type`.

>cluster.remote_store.index.restrict.async-durability is the existing example I can find where "restrict" is used. Why use "force" here?

This came up in review comment https://github.com/opensearch-project/OpenSearch/pull/11583#discussion_r1424521861. I think both `force` and `restrict` can be used, but I would prefer to keep `force`. Let me know if you think strongly on using `restrict`.

>Using cluster.indices.replication.strategy as a precedent, this should probably replace replication_type with replication.type.

Updated.

Reply:
We do have previous instances of force  - `cluster.routing.allocation.awareness.force.zone.values` and as a user - `force` sounds more explicit than `restrict` which could mean multiple things. 
I do not have strong opinions and we can go the other way for uniformity.